libc/src/__support/CPP/mutex.h

```diff
@@ -0,0 +1,47 @@
+//===--- A simple lock_guard implementation ---------------------*- C++ -*-===//
```

Comment:
```suggestion
//===--- self contained equivalent of std:: mutex.h ---------------------*- C++ -*-===//
```
(may need to adjust line length)